src/fheroes2/ai/ai_battle_spell.cpp

```diff
@@ -657,3 +658,58 @@ AI::SpellcastOutcome AI::BattlePlanner::spellSummonValue( const Spell & spell, c
 
     return bestOutcome;
 }
+
+AI::SpellcastOutcome AI::BattlePlanner::spellDragonSlayerValue( const Spell & spell, const Battle::Units & friendly, const Battle::Units & enemies ) const
+{
+    assert( spell.GetID() == Spell::DRAGONSLAYER );
+
+    double enemyArmyStrength = 0;
+    double dragonsStrength = 0;
+
+    int32_t numOfSlotsWithEnemyDragons = 0;
+
+    for ( const Battle::Unit * unit : enemies ) {
+        assert( unit != nullptr );
```

Comment:
Hi @ihhub I would add the `isValid()` check here as well - just in case:

```suggestion
        assert( unit != nullptr && unit->isValid() );
```